naivedb.go

```diff
@@ -0,0 +1,66 @@
+package naivedb
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type ReaderStringWriter interface {
+	io.Reader
+	io.StringWriter
+}
+
+type NaiveDB struct {
+	reader ReaderStringWriter
+}
+
+type FileBackedNaiveDB struct {
```

Comment:
Should I implement `FileBackedNaiveDB` in a different file? That way this file can focus entirely on NaiveDB.

Reply:
You'll probably eventually want to split it out if this file gets too large or complex, but there's nothing wrong with starting out with a single file for each package. Everything in the same package ends up in the same namespace regardless of which file it's in, so the main reason for splitting out code into multiple files is to keep things conceptually easier for people trying to read the code.